corelib-definitions/src/main/java/eu/europeana/corelib/definitions/solr/model/Query.java

```diff
@@ -251,20 +255,106 @@ public Query setFacets(List<String> facets) {
 	 * because it is a special query facet
 	 */
 	private void replaceSpecialFacets() {
-		List<String> additionalFacets = new ArrayList<String>();
-		for (String facet : facets) {
-			if (StringUtils.equals("DEFAULT", facet)) {
-				additionalFacets.addAll(new ArrayList<String>(defaultFacets));
-			} else if (StringUtils.equals("REUSABILITY", facet)) {
-				// skip it
-			} else {
-				additionalFacets.add(facet);
-			}
-		}
-		facets = additionalFacets;
+    boolean ok = false;
```

Comment:
use the java 7 string switch instead and use the default for return 0
http://docs.oracle.com/javase/7/docs/technotes/guides/language/strings-switch.html